### PR TITLE
fix(storybook): add a proxy for the request of AI Promts

### LINKED ISSUE #26521: [UI] Create a Middleware to use Storybook with dotCMS
As a developer, I want to use the plugin(com.dotcms.ai) of dotCMS to generate images and content with the Block Editor and Storyblock.

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* add a proxy to Storybook to use dotCMS as backend in the AI Extensions

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### <samp>🤖 Generated by Copilot at 7e8899e</samp>

This pull request adds proxy middleware to the storybook configuration to enable communication with the backend for the OpenAI integration. It creates a new file `middleware.js` that defines the proxy rules, modifies the `main.js` file to use the middleware, and installs the `http-proxy-middleware` package as a dev dependency.

## Related Issue
Fixes #26521 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 7e8899e</samp>

*  Add proxy middleware to storybook configuration to enable communication with backend for OpenAI features ([link](https://github.com/dotCMS/core/pull/26524/files?diff=unified&w=0#diff-c82c8893dc8aececc5879df411b5bd61f72926e31ee4202501df37b2e320cf6fL37-R38), [link](https://github.com/dotCMS/core/pull/26524/files?diff=unified&w=0#diff-4008ba1b3531ad1623a5d7f4d9752efb3ec977b0fe6811a314d7675c76f4b571R1-R45), [link](https://github.com/dotCMS/core/pull/26524/files?diff=unified&w=0#diff-a872cae44fcb0e8b8d97b9def0535e9fc9c7ee92a2d192e152b8db050ae00c08R190))

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]
Is there a way to alert us that we should run the backend to make this work properly? 

Sometimes we just run storybook standalone, so it would be nice to know why is failing.

**Review by manuelrojas — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/.storybook/main.js [hunk 1]
```diff
@@ -34,5 +34,6 @@ module.exports = {
         });
 
         return config;
-    }
+    },
+    middleware: './middleware.js'
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/.storybook/main.js:38]: Does this means that we need to run a dotCMS in localhost to run the stories?

**Review by fmontes — DISMISSED** [review]
If we need to run a dotcms to run storybook it will increase the complexity of having a storybook running which I don't think we want... did you consider mocks?

File: core-web/apps/dotcms-ui/.storybook/main.js (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/.storybook/main.js:38]: I will try with mocks. Now with Postman and a docker with 23.06, I can see the endpoint shape.

**Pull request merged**